src/runtimes/node/bundler.js

```diff
@@ -56,16 +48,47 @@ const bundleJsFile = async function ({
       platform: 'node',
       plugins: supportsAsyncAPI ? plugins : [],
       resolveExtensions: ['.js', '.jsx', '.mjs', '.cjs', '.ts', '.json'],
+      sourcemap: Boolean(config.nodeSourcemap),
+      sourceRoot: dirname(bundlePath),
       target: [nodeTarget],
     })
-    const inputs = Object.keys(data.metafile.inputs).map((path) => resolve(path))
+    const sourcemapPath = getSourcemapPath(metafile.outputs)
+    const inputs = Object.keys(metafile.inputs).map((path) => resolve(path))
+    const cleanTempFiles = getCleanupFunction(bundlePath, sourcemapPath)
 
-    return { bundlePath, cleanTempFiles, inputs, nativeNodeModules, warnings: data.warnings }
+    return {
+      bundlePath,
+      cleanTempFiles,
+      inputs,
+      nativeNodeModules,
+      sourcemapPath,
+      warnings,
+    }
   } catch (error) {
     error.customErrorInfo = { type: 'functionsBundling', location: { functionName: name } }
 
     throw error
   }
 }
 
+const getCleanupFunction = (bundlePath, sourcemapPath) => async () => {
+  try {
+    await pUnlink(bundlePath)
+
+    if (sourcemapPath) {
+      await pUnlink(sourcemapPath)
```

Comment:
[sand] As a tiny performance optimization, should this be done in parallel. Also, should the function be generalized so it can be re-used? Something along the lines of:

```js
const getCleanupFunction = (...paths) => async () => {
  await Promise.all(paths.filter(Boolean).map(cleanupFile))
}

const cleanupFile = async (path) => {
  try {
    await pUnlink(path)
  } catch (_) {}
}
```

Reply:
I did this in https://github.com/netlify/zip-it-and-ship-it/pull/447/commits/d9260d3bed870aed02eb4a1a63681e0f6ba02db0. Does that look okay to you?

Reply:
Yes! :+1:

Reply:
Great! Thanks for the tip.